### ISSUE TITLE
Using legend instead of label causes interaction and accessibility issues

### ISSUE THREAD

**Issue by robcresswell** [bootstrap-vue/bootstrap-vue#1425]
All of the forms seem to be using `<legend>` elements instead of `<label>`; this removes the expected click handling when you hit a label and it selects the field. Using `<label>` is quite useful on mobile screens since it makes the target interaction area bigger. You can see this on the form docs page: https://bootstrap-vue.js.org/docs/components/form

Is there any reason why this was done? If you check the Bootstrap documentation they seem to consistently use `<label>` and the click interaction works as expected.

I'd actually imagine this causes other accessibility issues too (since label for and input id are often expected) but I haven't tested it manually with a screen reader yet. Chrome's Lighthouse Accessibility Audit flags the lack of labels and deducts points for it.

**Comment by robcresswell** [comment]
Sorry for the spam, GitHubs formatting (and Slacks resolving of it) was causing issues.

**Comment by tmorehouse** [comment]
`label` elements only provide click interactivity of an input when they are associated with an input (i.e via the `for` attribute or via wrapping the label around a single input).

radio and check boxes (as well as teh custom file input) have built in labels, making the click area larger. Using `label` once again in the form group can be even more confusing for ARIA assistive technology users (i.e screen readers).

ARIA-wise, legends, by nature of markup, are associated with the contents of the fieldset, and provides a good compromise when its possible to have multiple inputs in a form-group (i.e. say a series of text boxes representing an address, or a bunch of radios or checkboxes)

In order to be useful, the label must be associated with the input (and only a single input). We originally had automatic setting (via the for` attribute, but it dodn't always work as by the time the form-group mountd (the child components might not be finished mounting), we sometimes couldn't find the input's ID and set the `for` attribute... making accessibility WORSE.  Trying to get around this, we set up DOM mutation observers to see when form inputs "appeared" inside the form-group. This added additional overhead, and fired mutation events whenever things changed in the form-group (i.e. showing feedback, adding/removing validation states, etc.  and not to mention, we didn't always pick the correct input with the form group (i.e. the form-group label might get associated with only a single checkbox in a checkbox group, which is not correct, as it should be associated with the outer wrapper of the form group, and not all people use the proper checkbox-group markup.)

So basically it is best if one sets the `for` attribute on the label manually.  Which is why we have PR #1423.  If the `label-for` property is set, we generate a `label` element, and if not set we generate a `legend` (which provides a good compromise, and works well with radios/checks and the bootstrap custom file input - all three of which have built in label elements).
